### PR TITLE
Level entities -- internal representation

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR relates to processing of entities from levels.txt. The goal is to use the required use cases implemented via PR #44 that allow to go from a Level representing a row from levels.txt to a specific concrete internal representation.

**Expected behavior:** 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- ~[ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")~
- ~[ ] Linked all relevant issues~
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Level.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class Level {
+
+    @NotNull
+    private final String levelId;
+
+    @NotNull
+    private final Float levelIndex;
+    private final String levelName;
+
+    public Level(@NotNull String levelId, @NotNull Float levelIndex, String levelName) {
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Level.java]: Make private to enforce Builder usage

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Level.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    public float getLevelIndex() {
+        return levelIndex;
+    }
+
+    public String getLevelName() {
+        return levelName;
+    }
+
+    @SuppressWarnings("CanBeFinal")
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Level.java]: Delete this line and add setter for `levelId` and `levelId`

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import java.util.List;
+
+/**
+ * Class for all entities defined in levels.txt. Can not be directly instantiated: user must use the
+ * {@code Level.LevelBuilder} to create this.
+ */
+public class Level extends GtfsEntity {
+    @NotNull
+    private final String levelId;
+    private final float levelIndex;
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java:34]: Missing nullability annotation?

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    public String getLevelName() {
+        return levelName;
+    }
+
+    /**
+     * Builder class to create {@link Level} objects. Allows an unordered definition of the different attributes of
+     * {@link Level}.
+     */
+    public static class LevelBuilder {
+        private String levelId;
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java:72]: Inconsistent nullability annotations, why is `levelName` the only annotated field?

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        }
+
+        /**
+         * Sets field levelIndex value and returns this
+         *
+         * @param levelIndex numeric index of the level that indicates relative position of this level in relation to other
+         *                   levels
+         * @return builder for future object creation
+         */
+        public LevelBuilder levelIndex(final Float levelIndex) {
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Level.java:96]: Parameter is not nullability annotated

File: domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/LevelTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+class LevelTest {
+
+    // Field levelId is annotated as `@NonNull` but test require this field to be null. Therefore annotation
+    // "@SuppressWarnings("ConstantConditions")" is used here to suppress lint.
+    @Test
+    public void createLevelWithNullLevelIdShouldGenerateNotice() {
+        final Level.LevelBuilder underTest = new Level.LevelBuilder();
+
+        //noinspection ConstantConditions
```
Comment (fabrice-v) [inline domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/LevelTest.java:35]: Missing explanatory comment

File: usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/port/GtfsDataRepository.java [hunk 7]
```diff
@@ -31,4 +32,8 @@
     Collection<Route> getRouteAll();
 
     Route getRouteById(final String routeId);
+
+    Level addLevel(final Level newLevel) throws IllegalArgumentException;
+
+    Level getLevelByLevelId(final String levelId);
```
Comment (fabrice-v) [inline usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/port/GtfsDataRepository.java:38]:
```suggestion
    Level getLevelById(final String levelId);
```

File: usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedLevelTest.java [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+class ProcessParsedLevelTest {
+
+    @Test
+    public void validatedParsedLevelShouldCreateLevelEntityAndBeAddedToGtfsDataRepository() {
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final GtfsDataRepository mockGtfsDataRepo = mock(GtfsDataRepository.class);
+        final Level.LevelBuilder mockBuilder = mock(Level.LevelBuilder.class, RETURNS_SELF);
+        final Level mockLevel = mock(Level.class);
+        final ParsedEntity mockParsedLevel = mock(ParsedEntity.class);
+        @SuppressWarnings("rawtypes") final EntityBuildResult mockGenericObject = mock(EntityBuildResult.class);
```
Comment (fabrice-v) [inline usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedLevelTest.java:46]:
```suggestion
        @SuppressWarnings("rawtypes") 
        final EntityBuildResult mockGenericObject = mock(EntityBuildResult.class);
```

Comment (lionel-nj) [reply usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedLevelTest.java:46]: Managed to remove the warning without using annotations.

**Pull request merged**